File: app/src/main/java/com/amarnehsoft/holyquran/SPController.java
package com.amarnehsoft.holyquran;

import android.content.Context;
import android.content.SharedPreferences;

import javax.inject.Inject;

public class SPController {
    private SharedPreferences sharedPreferences;

    @Inject
    public SPController(SharedPreferences sharedPreferences){
        this.sharedPreferences = sharedPreferences;
    }

    public void setAyaNumber(int number){
        sharedPreferences.edit().putInt("ayaNumber", number).apply();
    }

    public int getAyaNumber(){
        return sharedPreferences.getInt("ayaNumber", 1);
    }
}

File: app/src/main/java/com/amarnehsoft/holyquran/SPController.kt
package com.amarnehsoft.holyquran

import android.content.SharedPreferences

import javax.inject.Inject

class SPController @Inject
constructor(private val sharedPreferences: SharedPreferences) {

    var ayaNumber: Int
        get() = sharedPreferences.getInt(AYA_NUMBER, 1)
        set(number) = sharedPreferences.edit().putInt(AYA_NUMBER, number).apply()

    var isQuranDownloaded: Boolean
        get() = sharedPreferences.getBoolean(IS_QURAN_DOWNLOADED, false)
        set(value) = sharedPreferences.edit().putBoolean(IS_QURAN_DOWNLOADED, value).apply()

    companion object {
        private const val AYA_NUMBER = "ayaNumber"
        private const val IS_QURAN_DOWNLOADED = "isQuranDownloaded"
    }
}
